### PR TITLE
list all topics on main page

### PR DESCRIPTION
**Pull request by Zakkiassi**

### CODE REVIEW

**Review by leekli — CHANGES_REQUESTED** [review]
Hey Zak,

Well done for getting the topics to load. The first coding ticket task is to display all the articles which this does not do yet just.

Please can you go back to this one and implement 'displaying all articles' (ticket 4) and then re submit your pull request once complete.

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <Header />
+      <Routes>
+        <Route path="/" element={<Topics />} />
```
Comment (yalicee) [inline src/App.js:19]: You have "<Topics>" has your home path of '/' but there is no link to view all articles in this component so I can't see all of your articles like it mentions in the ticket

File: src/components/Topic.jsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { useState } from "react"
+
+
+const Topic = () => {
+    const [topic, setTopic] = useState([])
+    const {}
```
Comment (yalicee) [inline src/components/Topic.jsx]: What is this doing?

File: src/components/Articles.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        setArticles(articles);
+      });
+  }, [topic]);
+
+  return (
+    <main>
+      <ul>
+        <h2>Today's Articles</h2>
+        {articles.map((article) => {
+          //console.log(article);
```
Comment (yalicee) [inline src/components/Articles.jsx]: remove comment

File: src/components/Articles.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+const Articles = () => {
+  const [articles, setArticles] = useState([]);
+  const { topic } = useParams();
+  console.log(topic);
```
Comment (yalicee) [inline src/components/Articles.jsx]: remove console.log

File: src/components/Article.jsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  useEffect(() => {
+    axios.get(`https://ncnewsyk.herokuapp.com/api/articles/${article_id}`).then(
+      ({ data: { article } }) => {
+        setArticle(article);
+      },
+      [article_id]
+    );
+  });
+  console.log(article);
```
Comment (yalicee) [inline src/components/Article.jsx]: remove console.log pls

File: src/components/Article.jsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+const Article = () => {
```
Comment (yalicee) [inline src/components/Article.jsx:5]: I see you have made a single Article component but there is no link to view a single article

**Review by yalicee — CHANGES_REQUESTED** [review]
Read the comments I have made so you can see what needs changing.
Also I have noticed that you have combined tickets 4, 5, and 6 which means there is a lot of additional requirements in this PR.

First of all I have noticed:
There are no links to home or single articles, only topics
There is not much responsive design
You do not have handling for non-existent ids for your individual article component. A simple 404 message is enough.

**Review by leekli — CHANGES_REQUESTED** [review]
Hi Zak, thanks for making the changes, well done so far I can see some functionality working here however it's not quite there, see my feedback and suggested changes below. You will need to submit a new pull request once these are done:

- On the page(s) which display all the articles, or display articles by topic: You don't need the body of the article there, it just needs be a short summary of info on each article. The user doesn't need to see the body of the article unless they specifically click on that article. I would recommend you just display things like: article title, article author, maybe the topic name it belongs to? maybe a comment count and vote count? It's up to you really but I would remove the article body when its listing all articles or all articles by topic.
- When I go into all articles ... I can no longer see the topics, I have to click back. A user using React shouldn't need to press the back button. I can see the topics load on your home page only. Maybe it's worth having a NavBar which renders on every page, and the user can click things such as 'Home' which takes them back to the route "/", other options such as 'all articles' or 'topics' where it takes them to a page showing the topics or just back to the home page... again it's entirely up to you, but currently it is not easy or obvious to me as a user how I go back to a list of topics without clicking back
- I can see your single articles by id pages load which is great... I would recommend putting more info on this page, basically you could add everything that is returned from the api call on the object such as author, date, comment count, vote count and so on.
- If I manually type in an article id which does not exist in the address bar (e.g. /articles/9999) it takes me to an almost blank page, implement some error handling logic so the user gets a simple 404 error message page if they put in an article_id which does not exist.

Hope that helps, there's alot of feedback there but it's because there are 3 tickets in 1 pull request I'm afraid.

**Review by leekli — APPROVED** [review]
Brilliant Zak, well done for sorting all these changes. I can see it now functions as I would expect it to as a user.

Please feel free to merge this branch.

A note for your next branch that you create and pull request - make sure you remove all future console logs as there's still a few in your code. You don't need do this now, please merge this branch, then sort the console logs separately on a new branch when you begin ticket 7 :)

**Pull request merged**